ecc_enclave/enclave/shim.h

```diff
@@ -227,6 +227,18 @@ void get_creator_name(char* msp_id,  // MSP id of organization to which transact
 // upgrade it according to what highlighted in note-3.
 extern int get_random_bytes(uint8_t* buffer, size_t length);
 
+// get_channel_id
```

Comment:
Do we need these functions? get_channel_id actually already exists as commented out function template above but when we last discussed the shim api this (and other functions like TxID, getTxTimestamp, getBinding ..) didn't really seem necessary as MVP.    Not that it per-se hurts but if we start adding we should clear what we aim for with extension, e.g., do we want to match the complete set of functionality of go sdk?   Also, a bit unfortunate to add should shortly _after_ we published Tech Preview where we claim to have stabilized the API!

Additionally, given that we will have to re-implement these functions to get them securely, I wouldn't add new functions without any need ...

Reply:
> Do we need these functions?

What we need is the mspid and channelid parameters.
There are at least two ways to achieve this: inject them in the enclave, or (what this PR implements) retrieve them from Fabric and (not yet implemented) set them after the parameters are unsealed.
The latter (the implemented one) should be the right way, since we can leverage the Fabric infrastructure.

> get_channel_id actually already exists as commented out function template

I didn't see that... removed!

> didn't really seem necessary as MVP

Other solution would require setting and passing the needed parameters from somewhere else, thus replicating functionality that Fabric already has. In this way, the problem is solved, and FPC is more compliant with Fabric.

> if we start adding we should clear what we aim for with extension

We are not aiming to match the Fabric shim. FPC requires (as far as I remember): mspid, channelid, signedproposal. All these can be taken from Fabric without reinventing the wheel.

> a bit unfortunate to add should shortly after we published Tech Preview where we claim to have stabilized the API!

If you mean _forwards-compatible_ stability, then I think we should specify that that's not our objective now. In the sense that, if we find good/convenient ways to be more compatible with Fabric, we are going to add the missing APIs.
If you mean _backwards-compatible_ stability, then this PR maintains the stability since it is only adding code/APIs.

> Additionally, given that we will have to re-implement these functions to get them securely

I have modified the implementation to take this into consideration. For both mspid and channelid, the idea is to _set-once_ then always get the value, thereby restricting malicious actions from the peer. In particular, on first `get`, the value is set directly from the Fabric shim. However, when we implement seal/unseal of parameters, then the expectation is to set the values directly (without calls to the Fabric shim).

The values are still unverified. This matches both the current master and the design. In fact, the verification is performed later by the enclave registry. So, the _set-once_ will simply maintain the status quo. If the parameters are good, then the enclave can be registered and used, otherwise it cannot.

Reply:
> What we need is the mspid and channelid parameters.
> There are at least two ways to achieve this: inject them in the enclave, or (what this PR implements) retrieve them from Fabric and (not yet implemented) set them after the parameters are unsealed.
> The latter (the implemented one) should be the right way, since we can leverage the Fabric infrastructure.

I don't see your argument about "leveraging the fabric infrastructure".  This infrastructure exists only on the untrusted side. So you can extract them there are at time of triggering the enclave initialization and then pass them in a single ECALL, no need for separate additional OCALLS? This is the same point i think also Marcus is making (and we had made in the past)

Reply:
> > get_channel_id actually already exists as commented out function template
> 
> I didn't see that... removed!

Given that you want to use 1-1 call from enclave to go sdk calls for these functions -- while i think this is the wrong architecture, there is only so much i will fight for a better approach^1 -- i will veto any addition of these insecure functions in `shim.h`.  `Shim.h` is the API for the chaincode and should only provide secure implementations.   We _could_ implement them securely, but in that case there is no point to do the 1-1 sdk mapping insecurely as you do right now and you would have to implement them based on getSignedProposal.  

^1: As mentioned as a stand-alone comment: while i like small commits, in this particular case i think it would be much better to decide on the ECALL/OCALL interface of the registration flows. small PRs are good if the bigger sw architecture is clear, but in this case it isn't 100% clear to me and then it's difficult to review ...